src/Tgstation.Server.Host/Components/Chat/Providers/DiscordProvider.cs

```diff
@@ -936,6 +939,40 @@ List<IEmbedField> BuildUpdateEmbedFields(
 			if (gitHubOwner == null || gitHubRepo == null)
 				return fields;
 
+			previousRevisionInformation ??= new Models.RevisionInformation();
+			previousRevisionInformation.ActiveTestMerges ??= new List<RevInfoTestMerge>();
+
+			revisionInformation.ActiveTestMerges ??= new List<RevInfoTestMerge>();
```

Comment:
Don't mutate the input DTOs here. Instead do a coalescing access on them into a temp variable.

i.e.
```cs
var previousTestMerges = (IEnumerable<TestMerge>)previousRevisionInformation?.ActiveTestMerges ?? Enumerable.Empty<TestMerge>();
var currentTestMerges = (IEnumerable<TestMerge>)revisionInformation.ActiveTestMerges ?? Enumerable.Empty<TestMerge>();
```